thonny/editors.py

```diff
@@ -905,6 +917,13 @@ def close_single_untitled_unmodified_editor(self):
         if len(editors) == 1 and not editors[0].is_modified() and not editors[0].get_filename():
             self._cmd_close_file()
 
+    def _cmd_goto_source_line(self):
+        editor = self.get_current_editor()
+        if editor:
+            line_no = simpledialog.askinteger(tr("Goto"), tr("Line number"))
+            if line_no:
+                editor.select_line(line_no)
```

Comment:
Have you tested that too big and negative integers don't crash?

Reply:
thonny's implementation already works with negative and too big numbers.
so i have not added an extra check.